guava-tests/test/com/google/common/io/BaseEncodingTest.java

```diff
@@ -51,6 +54,34 @@ public void testSeparatorsExplicitly() {
     testEncodes(base32().withSeparator("*", 4), "foobar", "MZXW*6YTB*OI==*====");
   }
 
+  public void testBase64UrlWithPlus() {
```

Comment:
I would turn these into members in `testBase64InvalidDecodings()`

Reply:
Yeah

 Your suggestions is exactily right.

Reply:
Hi There! It looks like you've added these to `testBase16InvalidDecodings` (whereas base64 would be more appropriate).

Given that `base64Url()` is a different decoder than `base64()`,  I would consider adding a number of new unit test functions, situated nearby the block of base64 tests, something like `testBase64Url()` and `testBase64UrlInvalidDecodings()`.

Reply:
Thanks, Nick 
Sorry for my negligence.
I pushed the amend change with your suggestions

Reply:
@nglorioso  Hi

Can you help to review the new path?

Thank you very much.

Reply:
Thanks for your patience! I'm going to make a copy of this commit internally (attributed to you), then vend that back out, hopefully by tomorrow, if not Monday.

Reply:
Thank you.

Best regards